### PR TITLE
Logging fix

### PR DESCRIPTION
**Pull request by SeanTConrad**
## Why are these changes needed?

If the user ran the scan with should-upload set to False, and then again to upload the files, the log from the first run was never uploaded.

Changes:
- Logs all are in ./results with a timestamp prefix
- At the end of the scan, if should-upload is true, all unuploaded logs are uploaded. Then marked with a "u_" prefix to mark that they were already uploaded.
- Cleaned up some logging. Added logs in the beginning to show what the configuration is, what the args were, and what the resulting configuration for that scan is.
- Made the log file for the current scan more explicit insteady of having output_dir + "log.txt" in multiple places

## Checks

- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: src/verinfast/config.py [hunk 1]
```diff
@@ -213,6 +217,10 @@ def __init__(self, cfg_path: str = None) -> None:
                     parent = curr_path.parent
                     curr_path = parent
         self.handle_config_file()
+        if self.config is not FileNotFoundError:
```
Comment (aylusltd) [inline src/verinfast/config.py:220]: this is redundant to 226

File: src/verinfast/utils/utils.py [hunk 2]
```diff
@@ -140,9 +140,15 @@ def chunks(lst, n):
...
-    def __init__(self, path: str, debug: bool = False):
+    def __init__(
+            self,
+            path: str = None,
+            file: str = None,
+            debug: bool = False):
         self.path = path
-        self.logFile = os.path.join(path, "log.txt")
+        self.file = file
+        if self.path is not None:
```
Comment (aylusltd) [inline src/verinfast/utils/utils.py:150]: what if both are None? in theory they're both optional now.

File: src/verinfast/utils/utils.py [hunk 3]
```diff
@@ -158,8 +164,15 @@ def log(self, msg, tag=None, display=False, timestamp=True):
         else:
             output = f"{msg}"
 
-        with open(self.logFile, 'a') as f:
-            f.write(output+"\n")
+        if self.path is not None:
```
Comment (aylusltd) [inline src/verinfast/utils/utils.py:167]: there should always be a file after instantiation or instantiation should fail.

**Review by aylusltd — CHANGES_REQUESTED** [review]

File: src/verinfast/config.py (hunk 1 above)
Comment (SeanTConrad) [reply src/verinfast/config.py:220]: It's not. On line 225, handle_args gets called which may modify self.config.

So orig_config stores the config from the file, but self.config ends up being a combination of what was in the file plus arg changes. I wanted to see both in the logs.

**Review by aylusltd — COMMENTED** [review]
github sucks

File: src/verinfast/config.py (hunk 1 above)
Comment (SeanTConrad) [reply src/verinfast/config.py:220]: You were correct.

File: src/verinfast/cloud/aws/costs.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 from verinfast.utils.utils import DebugLog
 from verinfast.cloud.aws.get_profile import find_profile
-debugLog = DebugLog(os.getcwd())
+debugLog = DebugLog(path=os.getcwd())
```
[the remark this answers is not in the document]
Comment (SeanTConrad) [reply src/verinfast/cloud/aws/costs.py]: done

File: src/verinfast/cloud/aws/get_profile.py [hunk 5]
```diff
@@ -5,7 +5,7 @@
 from verinfast.utils.utils import DebugLog
 
 
-debugLog = DebugLog(os.getcwd())
+debugLog = DebugLog(path=os.getcwd())
```
[the remark this answers is not in the document]
Comment (SeanTConrad) [reply src/verinfast/cloud/aws/get_profile.py]: done

File: src/verinfast/utils/utils.py [hunk 6]
```diff
@@ -139,10 +139,28 @@ def chunks(lst, n):
...
-    def __init__(self, path: str, debug: bool = False):
-        self.path = path
-        self.logFile = os.path.join(path, "log.txt")
+    def __init__(
+            self,
+            path: str = None,
+            file: str = None,
+            debug: bool = False):
+        if path is None and file is None:
+            print("DebugLog: No log file or path specified")
```
Comment (aylusltd) [inline src/verinfast/utils/utils.py:154]: should raise an exception here, don't let it through.

File: src/verinfast/utils/utils.py [hunk 7]
```diff
@@ -139,10 +139,28 @@ def chunks(lst, n):
...
+            return
+        if path is not None and file is not None:
+            print("DebugLog: Both log file and path specified")
```
Comment (aylusltd) [inline src/verinfast/utils/utils.py:157]: same, raise the exception and terminate

File: src/verinfast/config.py [hunk 8]
```diff
@@ -256,8 +260,21 @@ def __init__(self, cfg_path: str = None) -> None:
             self.upload_conf.code_separator = self.server_code_separator
 
         os.makedirs(self.output_dir, exist_ok=True)
-        debugLog = DebugLog(path=self.output_dir)
-        debugLog.log(msg=self.config, tag="Config", display=True)
+        self.log_file = os.path.join(
```
Comment (aylusltd) [inline src/verinfast/config.py:263]: this is redundant

File: src/verinfast/agent.py [hunk 9]
```diff
@@ -874,10 +869,25 @@ def main():
...
-        os.unlink(agent.config.output_dir+"/log.txt")
-        print(f"""The log for this run has moved to:
-              {d}/{new_folder_name}/{new_file_name}""")
+        shutil.copy2(agent.config.log_file, fp)
+        print(f"""The log for this run was copied to:
+            {d}/{new_folder_name}/{new_file_name}""")
+
+    # If user opts to upload results, upload and non-uploaded logs
+    if agent.config.shouldUpload:
+        log_list = os.listdir(agent.config.output_dir)
```
Comment (aylusltd) [inline src/verinfast/agent.py:878]: you should scan the home dir where files are copied. That's the safe spot.

**Review by aylusltd — APPROVED** [review]

**Pull request merged**